### ISSUE TITLE
Not a big deal, but this block is probably not required. And another question on dependencies.

### ISSUE THREAD

**Issue by scm573** [googlesamples/android-instant-apps#26]
As title. I commented out this block and everything still goes fine. (Android Studio 3.0 Beta 2)

https://github.com/googlesamples/android-instant-apps/blob/39b51376a71b3c3485dd3fb547f587c70bc180f9/hello-feature-module/features/hello/build.gradle#L37-L43

-------------------------------------------------------

And actually I'm facing that problem in my own project:
> Android dependency 'com.android.support:support-v4' has different version for the compile (21.0.3) and runtime (26.0.1) classpath. You should manually set the same version via DependencyResolution

What I'm trying is to refactor my app to an instant app. In my base-feature module, since I'm using data binding, which the 21.0.3 comes from, the sync error pasted above will occur if I add support-v4 dependency explicitly:
`implementation 'com.android.support:support-v4:26.0.1'` 

If I don't do so and exclude the support-v4 module from other dependencies, there will be no sync error but those methods later than support-v4:21.0.3 cannot be resolved.

So one possible solution I'm considering currently is to exclude the support-v4 module from data binding itself, but I have no idea how to do that.
It would be greatly appreciated if anyone could help me out of this trouble. Many thanks.

**Comment by keyboardsurfer** [comment]
You can try to use [ResolutionStrategy and force](https://docs.gradle.org/3.4.1/dsl/org.gradle.api.artifacts.ResolutionStrategy.html) a specific version of the support library.

In the sample's case I'm happy with the exclusion for now.

**Comment by scm573** [comment]
@keyboardsurfer 
Thanks for the help. I forced the version of support-v4 using this block in root build.gradle:
```groovy
subprojects {
    project.configurations.all {
        resolutionStrategy.eachDependency { details ->
            if (details.requested.group == 'com.android.support'
                    && !details.requested.name.contains('multidex') ) {
                details.useVersion "$supportlib_version"
            }
        }
    }
}
```
The sync works fine but when rebuild the project this error will appear:
> android.databinding.tool.util.LoggedErrorException: Found data binding errors.
Cannot find the setter for attribute 'android:onClick' with ...

I checked the binding classes and it seems that they were generated correctly. Do you have any idea on this?

**Comment by keyboardsurfer** [comment]
This is out of scope for this sample. I recommend posting to stackoverflow.

**Comment by scm573** [comment]
Yes you're right. Anyway, thanks very much.

**Comment by keyboardsurfer** [comment]
Feel free to link to the stackoverflow question, once it's there.

**Comment by scm573** [comment]
Finally I got out of the build error (thanks to the help from Google Developer Support):

> android.databinding.tool.util.LoggedErrorException: Found data binding errors.
Cannot find the setter for attribute 'android:onClick' with ... (here I have a FAB)

I'll just leave my solution here for those who're facing the same problem.

So the root cause was when I refactored my project in Android Studio 3.0, I modified all of the "compile" dependency declaration to "implementation", as suggested by the IDE. However "implementation" does not expose the dependency to other modules that depend on it (which can be referred here: https://stackoverflow.com/questions/44493378/whats-the-difference-between-implementation-and-compile-in-gradle/44493379#44493379), as the result my FAB which comes from the design support library couldn't be resolved.

The solution is simply using "api" declaration instead of "implementation". And the build will work well. Yay~
